locale/Makefile

```
@@ -6,8 +6,8 @@
 # make tx-update - download and merge translations from Transifex
 # make clean - clean everything
 #
-DOMAIN = foreman_plugin_template
-VERSION = $(shell ruby -e 'require "rubygems";spec = Gem::Specification::load(Dir.glob("../*.gemspec")[0]);puts spec.version')
+DOMAIN = $(shell ruby -rrubygems -e 'puts Gem::Specification::load(Dir.glob("../*.gemspec")[0]).name')
```

Alternative:
```suggestion
DOMAIN = $(shell basename -s .gemspec ../*gemspec)
```

There may even be some fancy logic in Make itself.

@evgeni thoughts?

---

Also, this breaks on `foreman-tasks` where the domain is `foreman_tasks`.

---

There are smarter ways to do `basename` in Make I think, but we really should not break tasks (which has that weird naming, but here we are).

The Ruby line seems reasonable to me.

---

The convention is that the filename also matches the gem name so it shouldn't be a difference. In foreman-tasks it's also `foreman-tasks.gemspec` so there it wouldn't be a difference: both would be wrong. It'd be so much better if we renamed the gem to `foreman_tasks`.

---

To be honest I don't have good experience with renaming gems so I'd like to avoid it if possible. Could we rename the domain instead?

---

No, you can't do that since `-` is not a valid character. It will fail [here](https://github.com/theforeman/foreman/blob/64a2ff82d9b7d3237ac59d0283af3a35b4533eb0/lib/tasks/gettext.rake#L26). I also don't know if the domain is used somewhere else.

Note that renaming the gem would also make it follow our other conventions and we don't need to special case it everywhere. Things like https://github.com/theforeman/puppet-foreman/blob/56404b20402960ff32ec983ce06c7c67517e2ce1/manifests/plugin/tasks.pp#L18 could be removed.

---

Compare [plugin_apipie](https://github.com/theforeman/foreman/blob/eaf0052b16ad622dcce16615728b687825794b50/lib/tasks/plugin_apipie.rake#L8-L9) to [gettext](https://github.com/theforeman/foreman/blob/eaf0052b16ad622dcce16615728b687825794b50/lib/tasks/gettext.rake#L25-L26).

I think the question is whether we want to support a dash at all. I think `foreman-tasks` is the only exception so I'm leaning to fixing the exception rather than allowing everything.

---

I've opened https://projects.theforeman.org/issues/35813 to track the foreman-tasks rename.